src/Microsoft.Azure.SignalR.Common/Exceptions/ExceptionExtensions.cs

```diff
@@ -35,5 +37,19 @@ internal static Exception WrapAsAzureSignalRException(this Exception e, Uri base
                     return e;
             }
         }
+
+        internal static Exception WrapAsAzureSignalRException(this Exception e, WebSocketConnectionOptions options)
+        {
+            switch (e)
+            {
+                case WebSocketException webSocketException:
+                    if (options.AuthType == AuthType.AzureAd && e.Message.Contains("401"))
```

Comment:
looks cool, can it be more specific using "StartWith"? Could it be also applied to other auth types?

Reply:
401 makes the most sense in that error message. If we use the `StartWith`, it could be `.StartWith("The server returned status code '401'"), which is no better than `Contains`.

Reply:
Is it possible that other place contains some random 401 number that is not related to status code?

Reply:
The chance is low. They doesn't contain random strings as I know.